opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/jms/JmsTracingTest.java

```diff
@@ -0,0 +1,81 @@
+package io.opentracing.contrib.spring.cloud.jms;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import io.opentracing.contrib.spring.cloud.MockTracingConfiguration;
+import io.opentracing.contrib.spring.cloud.TestUtils;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author <a href="mailto:ales.justin@jboss.org">Ales Justin</a>
+ */
+@SpringBootTest(
+  webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+  classes = {MockTracingConfiguration.class, JmsTestConfiguration.class, JmsTracingTest.MsgController.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+public class JmsTracingTest {
+  @LocalServerPort
+  int port;
```

Comment:
OK, will check that starter.

Yeah, Sleuth has no JMS atm.
The abstraction covers other messaging impls, and the idea is to also add a higher level abstraction for JMS as well, so existing Sleuth instrumentation would cover JMS ootb.